src/app/flows/fines/fines-mac/fines-mac-offence-details/fines-mac-offence-details-add-an-offence/fines-mac-offence-details-add-an-offence.component.ts

```diff
@@ -143,6 +145,36 @@ export class FinesMacOffenceDetailsAddAnOffenceComponent
     }
   }
 
+  /**
+   * Retrieves the collection order date from the payment terms.
+   *
+   * @returns {Date | null} The collection order date if available, otherwise null.
+   */
+  private getCollectionOrderDate(): Date | null {
+    const { formData: paymentTerms } = this.finesService.finesMacState.paymentTerms;
+    if (paymentTerms?.fm_payment_terms_collection_order_date) {
+      return this.dateService.getDateFromFormat(paymentTerms.fm_payment_terms_collection_order_date, 'dd/MM/yyyy');
+    }
+    return null;
+  }
+
+  /**
+   * Checks the payment terms collection order by comparing the collection order date
+   * with the earliest date of sentence. If the collection order date is earlier than
+   * the earliest date of sentence, it sets the payment terms status to INCOMPLETE.
+   *
+   * @private
+   * @returns {void}
+   */
+  private checkPaymentTermsCollectionOrder(): void {
+    const collectionOrderDate = this.getCollectionOrderDate();
+    const earliestDateOfSentence = this.finesService.getEarliestDateOfSentence();
+
+    if (collectionOrderDate && earliestDateOfSentence && collectionOrderDate < earliestDateOfSentence) {
```

Comment:
Consider splitting into  separate variables to make the if statement easier to read.

e.g.
hasCollectionOrderEarliestDate = collectionOrderDate && earliestDateOfSentence
isCollectionOrderLessThanEarliestDate =  collectionOrderDate < earliestDateOfSentence

if(hasCollectionOrderEarliestDate && isCollectionOrderLessThanEarliestDate) {}